osu.Game/Rulesets/Scoring/Score.cs

```diff
@@ -28,11 +29,26 @@ public class Score
         public int Combo { get; set; }
 
         [JsonProperty(@"mods")]
-        protected string[] ModStrings { get; set; } //todo: parse to Mod objects
+        private string[] modStrings { get; set; }
 
-        public RulesetInfo Ruleset { get; set; }
+        public RulesetInfo Ruleset;
 
-        public Mod[] Mods { get; set; }
+        private Mod[] mods;
+        public Mod[] Mods
+        {
+            get
+            {
+                // Evaluate the mod strings if necessary
+                return mods ?? (mods = Ruleset.CreateInstance().GetAllMods().Where(mod => modStrings.Contains(mod.ShortenedName)).ToArray());

```

Comment:
I'm afraid parsing on its setter is not possible as the ruleset isn't set at this time. The way I had it before was that I parsed it in the ruleset's setter.